src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java

```diff
@@ -1,16 +1,29 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.MovieDao;
+import mate.academy.hibernate.relations.dao.impl.MovieDaoImpl;
+import mate.academy.hibernate.relations.exception.DataProcessingException;
 import mate.academy.hibernate.relations.model.Movie;
 import mate.academy.hibernate.relations.service.MovieService;
+import mate.academy.hibernate.relations.util.HibernateUtil;
+import org.hibernate.SessionFactory;
 
 public class MovieServiceImpl implements MovieService {
+    private SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+    private MovieDao movieDao = new MovieDaoImpl(sessionFactory);
+
+    public MovieServiceImpl(MovieDao movieDao) {
+        this.movieDao = movieDao;
+    }
+
     @Override
     public Movie add(Movie movie) {
-        return null;
+        return movieDao.add(movie);
     }
 
     @Override
     public Movie get(Long id) {
-        return null;
+        return movieDao.get(id).orElseThrow(
+                () -> new DataProcessingException("Failed to get movie with id: " + id));
```

Comment:
You are throwing a `DataProcessingException` here. It's better to handle this exception on a DAO layer and return an Optional from DAO. Then you can use `orElseThrow()` without arguments, it will throw `NoSuchElementException` if value is not present, which is more appropriate in this case.